packages/web/src/components/download-track-archive-modal/DownloadTrackArchiveModal.tsx

```diff
@@ -0,0 +1,138 @@
+import { useCallback, useEffect } from 'react'
+
+import {
+  useCancelStemsArchiveJob,
+  useDownloadTrackStems,
+  useGetStemsArchiveJobStatus
+} from '@audius/common/api'
+import { useDownloadTrackArchiveModal } from '@audius/common/store'
+import {
+  Modal,
+  ModalContent,
+  ModalHeader,
+  Text,
+  Flex,
+  LoadingSpinner,
+  IconFolder,
+  ModalTitle,
+  IconReceive,
+  Hint,
+  IconError,
+  TextLink
+} from '@audius/harmony'
+
+import { env } from 'services/env'
+
+const messages = {
+  title: 'Preparing Download',
+  zippingFiles: (count: number) => `Zipping files (${count})`,
+  error: 'Something went wrong. Please check your connection and try again.',
+  tryAgain: 'Try again.'
+}
+
+const triggerDownload = (url: string) => {
+  if (document) {
+    const link = document.createElement('a')
+    link.href = url
+    link.click()
+    link.remove()
+  } else {
+    throw new Error('No document found')
+  }
+}
+
+export const DownloadTrackArchiveModal = () => {
+  const {
+    data: { trackId, fileCount },
+    isOpen,
+    onClose,
+    onClosed
+  } = useDownloadTrackArchiveModal()
+
+  const {
+    mutate: downloadTrackStems,
+    isError: initiateDownloadFailed,
+    isPending: isStartingDownload,
+    data: { id: jobId } = {}
+  } = useDownloadTrackStems({
+    trackId
+  })
+
+  const { mutate: cancelStemsArchiveJob } = useCancelStemsArchiveJob()
+
+  const { data: jobState } = useGetStemsArchiveJobStatus({
+    jobId
+  })
+
+  const hasError =
+    !isStartingDownload &&
+    (initiateDownloadFailed || jobState?.state === 'failed')
+
+  useEffect(() => {
+    if (isOpen) {
+      downloadTrackStems()
+    }
+  }, [isOpen, downloadTrackStems, trackId])
+
+  useEffect(() => {
+    if (jobState?.state === 'completed') {
+      triggerDownload(`${env.ARCHIVE_ENDPOINT}/archive/stems/download/${jobId}`)
+      onClose()
+    }
+  }, [jobState, onClose, jobId])
+
+  const handleClose = useCallback(() => {
+    if (jobId) {
+      cancelStemsArchiveJob({ jobId })
+    }
+    onClose()
+  }, [onClose, jobId, cancelStemsArchiveJob])
+
+  const handleRetry = useCallback(() => {
+    downloadTrackStems()
```

Comment:
if isError, are we sure the archivejob is cancelled?

Reply:
Oh good catch, this was originally only running if the job state returned back had an error status. I'll think through this again. It's probably alright to start over here, since the server has logic to handle duplicate requests for the same trackId/userId combo and just return the state of the existing job.